# ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã‚¬ã‚¤ãƒ‰

## ç›®æ¬¡

1. [ã¯ã˜ã‚ã«](#ã¯ã˜ã‚ã«)
2. [ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã®å„ªå…ˆé †ä½](#ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã®å„ªå…ˆé †ä½)
3. [èªè¨¼é–¢é€£](#èªè¨¼é–¢é€£)
   - 3.1 [ãƒ­ã‚°ã‚¤ãƒ³ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã®å¤‰æ›´](#31-ãƒ­ã‚°ã‚¤ãƒ³ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã®å¤‰æ›´)
   - 3.2 [èªè¨¼ API ã‚¨ãƒ³ãƒ‰ãƒã‚¤ãƒ³ãƒˆ](#32-èªè¨¼apiã‚¨ãƒ³ãƒ‰ãƒã‚¤ãƒ³ãƒˆ)
   - 3.3 [ãƒ¦ãƒ¼ã‚¶ãƒ¼å‹å®šç¾©](#33-ãƒ¦ãƒ¼ã‚¶ãƒ¼å‹å®šç¾©)
   - 3.4 [èªè¨¼å¾Œã®ãƒªãƒ€ã‚¤ãƒ¬ã‚¯ãƒˆå…ˆ](#34-èªè¨¼å¾Œã®ãƒªãƒ€ã‚¤ãƒ¬ã‚¯ãƒˆå…ˆ)
   - 3.5 [ä¿è­·ãƒ«ãƒ¼ãƒˆã®è¨­å®š](#35-ä¿è­·ãƒ«ãƒ¼ãƒˆã®è¨­å®š)
   - 3.6 [Cookie/Token è¨­å®š](#36-cookietokenè¨­å®š)
4. [è‰²ãƒ»ãƒ†ãƒ¼ãƒ](#4-è‰²ãƒ†ãƒ¼ãƒ)
   - 4.1 [ã‚°ãƒ­ãƒ¼ãƒãƒ« CSS ã®è‰²å®šç¾©](#41-ã‚°ãƒ­ãƒ¼ãƒãƒ«cssã®è‰²å®šç¾©)
   - 4.2 [Tailwind è¨­å®š](#42-tailwindè¨­å®š)
   - 4.3 [shadcn/ui ã®ãƒ†ãƒ¼ãƒ](#43-shadcnuiã®ãƒ†ãƒ¼ãƒ)
5. [ãƒšãƒ¼ã‚¸ãƒ»ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ](#5-ãƒšãƒ¼ã‚¸ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ)
   - 5.1 [ãƒ­ã‚°ã‚¤ãƒ³ãƒšãƒ¼ã‚¸](#51-ãƒ­ã‚°ã‚¤ãƒ³ãƒšãƒ¼ã‚¸)
   - 5.2 [ãƒ€ãƒƒã‚·ãƒ¥ãƒœãƒ¼ãƒ‰ãƒšãƒ¼ã‚¸](#52-ãƒ€ãƒƒã‚·ãƒ¥ãƒœãƒ¼ãƒ‰ãƒšãƒ¼ã‚¸)
   - 5.3 [èªè¨¼ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ](#53-èªè¨¼ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ)
6. [ç’°å¢ƒè¨­å®š](#6-ç’°å¢ƒè¨­å®š)
   - 6.1 [ã‚¢ãƒ—ãƒªåãƒ»ãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿](#61-ã‚¢ãƒ—ãƒªåãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿)
   - 6.2 [ç’°å¢ƒå¤‰æ•°](#62-ç’°å¢ƒå¤‰æ•°)
   - 6.3 [JWT éµã®è¨­å®š](#63-jwtéµã®è¨­å®š)
   - 6.4 [CORS è¨­å®š](#64-corsè¨­å®š)
   - 6.5 [ãƒ­ã‚´ãƒ»ãƒ•ã‚¡ãƒ“ã‚³ãƒ³](#65-ãƒ­ã‚´ãƒ•ã‚¡ãƒ“ã‚³ãƒ³)
   - 6.6 [package.json](#66-packagejson)
7. [ã‚¤ãƒ³ãƒ•ãƒ©è¨­å®š](#7-ã‚¤ãƒ³ãƒ•ãƒ©è¨­å®š)
   - 7.1 [CDK ã‚·ã‚¹ãƒ†ãƒ å](#71-cdkã‚·ã‚¹ãƒ†ãƒ å)
   - 7.2 [Docker Compose](#72-docker-compose)
8. [ãƒã‚§ãƒƒã‚¯ãƒªã‚¹ãƒˆ](#8-ãƒã‚§ãƒƒã‚¯ãƒªã‚¹ãƒˆ)

---

## ã¯ã˜ã‚ã«

ã“ã®ãƒ†ãƒ³ãƒ—ãƒ¬ãƒ¼ãƒˆã¯æ±ç”¨çš„ãªè¨­å®šã§æ§‹ç¯‰ã•ã‚Œã¦ã„ã¾ã™ã€‚æ–°ã—ã„ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆã‚’é–‹å§‹ã™ã‚‹éš›ã¯ã€ã“ã®ã‚¬ã‚¤ãƒ‰ã«å¾“ã£ã¦ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆå›ºæœ‰ã®è¨­å®šã«ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã—ã¦ãã ã•ã„ã€‚

### ãƒ†ãƒ³ãƒ—ãƒ¬ãƒ¼ãƒˆã®ç¾çŠ¶

- âœ… èªè¨¼æ©Ÿèƒ½ï¼ˆãƒ­ã‚°ã‚¤ãƒ³ãƒ»ãƒ­ã‚°ã‚¢ã‚¦ãƒˆï¼‰ãŒå®Ÿè£…æ¸ˆã¿
- âœ… ãƒ€ãƒƒã‚·ãƒ¥ãƒœãƒ¼ãƒ‰ãƒšãƒ¼ã‚¸ãŒå®Ÿè£…æ¸ˆã¿
- âš ï¸ ãƒ‡ã‚¶ã‚¤ãƒ³ãƒ»è‰²ã¯ä»®ã®è¨­å®š
- âš ï¸ èªè¨¼ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã¯æ±ç”¨çš„ãªè¨­å®šï¼ˆ`login_id`ï¼‰
- âš ï¸ ã‚¢ãƒ—ãƒªåã‚„ãƒ­ã‚´ã¯ä»®ã®ã‚‚ã®

---

## ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã®å„ªå…ˆé †ä½

### ğŸ”´ å¿…é ˆï¼ˆé«˜å„ªå…ˆåº¦ï¼‰- ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆé–‹å§‹æ™‚ã«å¿…ãšå¤‰æ›´

| é …ç›®                                                 | ä½œæ¥­æ™‚é–“ | é‡è¦åº¦ |
| ---------------------------------------------------- | -------- | ------ |
| [èªè¨¼ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã®å¤‰æ›´](#31-ãƒ­ã‚°ã‚¤ãƒ³ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã®å¤‰æ›´) | 30 åˆ†    | â˜…â˜…â˜…    |
| [ã‚¢ãƒ—ãƒªåãƒ»ãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿](#61-ã‚¢ãƒ—ãƒªåãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿)       | 15 åˆ†    | â˜…â˜…â˜…    |
| [ç’°å¢ƒå¤‰æ•°ã®è¨­å®š](#62-ç’°å¢ƒå¤‰æ•°)                       | 30 åˆ†    | â˜…â˜…â˜…    |
| [CORS è¨­å®š](#64-corsè¨­å®š)                            | 10 åˆ†    | â˜…â˜…â˜…    |
| [JWT éµã®ç”Ÿæˆ](#63-jwtéµã®è¨­å®š)                      | 15 åˆ†    | â˜…â˜…â˜…    |
| [ãƒ­ã‚´ãƒ»ãƒ•ã‚¡ãƒ“ã‚³ãƒ³](#65-ãƒ­ã‚´ãƒ•ã‚¡ãƒ“ã‚³ãƒ³)               | 20 åˆ†    | â˜…â˜…â˜…    |

**åˆè¨ˆ**: ç´„ 2 æ™‚é–“

---

### ğŸŸ¡ é‡è¦ï¼ˆä¸­å„ªå…ˆåº¦ï¼‰- åˆæœŸé–‹ç™ºæ®µéšã§å¯¾å¿œ

| é …ç›®                                                     | ä½œæ¥­æ™‚é–“ | é‡è¦åº¦ |
| -------------------------------------------------------- | -------- | ------ |
| [è‰²ãƒ»ãƒ†ãƒ¼ãƒã®å¤‰æ›´](#4-è‰²ãƒ†ãƒ¼ãƒ)                          | 1 æ™‚é–“   | â˜…â˜…â˜†    |
| [ãƒ€ãƒƒã‚·ãƒ¥ãƒœãƒ¼ãƒ‰ã®ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚º](#52-ãƒ€ãƒƒã‚·ãƒ¥ãƒœãƒ¼ãƒ‰ãƒšãƒ¼ã‚¸) | 2-4 æ™‚é–“ | â˜…â˜…â˜†    |
| [ãƒ­ã‚°ã‚¤ãƒ³ãƒšãƒ¼ã‚¸ãƒ‡ã‚¶ã‚¤ãƒ³](#51-ãƒ­ã‚°ã‚¤ãƒ³ãƒšãƒ¼ã‚¸)             | 1-2 æ™‚é–“ | â˜…â˜…â˜†    |
| [ä¿è­·ãƒ«ãƒ¼ãƒˆã®è¿½åŠ ](#35-ä¿è­·ãƒ«ãƒ¼ãƒˆã®è¨­å®š)                 | 30 åˆ†    | â˜…â˜…â˜†    |
| [CDK ã‚·ã‚¹ãƒ†ãƒ å](#71-cdkã‚·ã‚¹ãƒ†ãƒ å)                      | 10 åˆ†    | â˜…â˜…â˜†    |

**åˆè¨ˆ**: ç´„ 5-8 æ™‚é–“

---

### ğŸŸ¢ æ¨å¥¨ï¼ˆä½å„ªå…ˆåº¦ï¼‰- å¿…è¦ã«å¿œã˜ã¦å¯¾å¿œ

| é …ç›®                                    | ä½œæ¥­æ™‚é–“ | é‡è¦åº¦ |
| --------------------------------------- | -------- | ------ |
| [ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆæ”¹å–„](#53-èªè¨¼ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ)    | 2-4 æ™‚é–“ | â˜…â˜†â˜†    |
| ãƒ‡ãƒãƒƒã‚°ãƒ­ã‚°å‰Šé™¤                        | 30 åˆ†    | â˜…â˜†â˜†    |
| [ãƒ‘ãƒƒã‚±ãƒ¼ã‚¸åå¤‰æ›´](#66-packagejson)     | 5 åˆ†     | â˜…â˜†â˜†    |
| [Docker ã‚³ãƒ³ãƒ†ãƒŠå](#72-docker-compose) | 10 åˆ†    | â˜…â˜†â˜†    |

---

## èªè¨¼é–¢é€£

### 3.1 ãƒ­ã‚°ã‚¤ãƒ³ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã®å¤‰æ›´

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹ 1: ãƒ¡ãƒ¼ãƒ«ã‚¢ãƒ‰ãƒ¬ã‚¹ã«å¤‰æ›´

**Step 1**: ãƒ•ã‚©ãƒ¼ãƒ ã‚’å¤‰æ›´

```tsx
// frontend/src/widgets/login-form/LoginForm.tsx
<Label htmlFor='email'>ãƒ¡ãƒ¼ãƒ«ã‚¢ãƒ‰ãƒ¬ã‚¹</Label>
<Input
  id='email'
  type='email'
  placeholder='email@example.com'
  {...register('email')}
/>
```

**Step 2**: ãƒãƒªãƒ‡ãƒ¼ã‚·ãƒ§ãƒ³ã‚’å¤‰æ›´

```tsx
// frontend/src/widgets/login-form/LoginForm.tsx
const loginSchema = z.object({
  email: z
    .string()
    .email("æœ‰åŠ¹ãªãƒ¡ãƒ¼ãƒ«ã‚¢ãƒ‰ãƒ¬ã‚¹ã‚’å…¥åŠ›ã—ã¦ãã ã•ã„")
    .min(1, "ãƒ¡ãƒ¼ãƒ«ã‚¢ãƒ‰ãƒ¬ã‚¹ã¯å¿…é ˆã§ã™"),
  password: z.string().min(8, "ãƒ‘ã‚¹ãƒ¯ãƒ¼ãƒ‰ã¯8æ–‡å­—ä»¥ä¸Šã§ã‚ã‚‹å¿…è¦ãŒã‚ã‚Šã¾ã™"),
});
```

**Step 3**: å‹å®šç¾©ã‚’å¤‰æ›´

```typescript
// frontend/src/entities/auth/model/types.ts
export interface LoginRequest {
  email: string; // â† login_id ã‹ã‚‰å¤‰æ›´
  password: string;
}
```

**Step 4**: ãƒãƒƒã‚¯ã‚¨ãƒ³ãƒ‰ã®ãƒ¢ãƒ‡ãƒ«ã‚’å¤‰æ›´

```python
# backend/app/domain/auth/model.py
@dataclass
class LoginRequest:
    email: str  # â† login_id ã‹ã‚‰å¤‰æ›´
    password: str
```

**Step 5**: ãƒãƒƒã‚¯ã‚¨ãƒ³ãƒ‰ã®ãƒ¦ãƒ¼ã‚¹ã‚±ãƒ¼ã‚¹ã‚’å¤‰æ›´

```python
# backend/app/application/auth/login_use_case.py
def execute(self, input_dto: LoginInputDto) -> LoginOutputDto:
    user = self.user_repository.find_by_email(input_dto.email)  # â† å¤‰æ›´
    if not user:
        raise ValueError("Invalid credentials")
    # ...
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹ 2: ãƒ¦ãƒ¼ã‚¶ãƒ¼åã«å¤‰æ›´

```tsx
// frontend/src/widgets/login-form/LoginForm.tsx
<Label htmlFor='username'>ãƒ¦ãƒ¼ã‚¶ãƒ¼å</Label>
<Input
  id='username'
  type='text'
  placeholder='ãƒ¦ãƒ¼ã‚¶ãƒ¼åã‚’å…¥åŠ›'
  {...register('username')}
/>
```

```typescript
// frontend/src/entities/auth/model/types.ts
export interface LoginRequest {
  username: string;
  password: string;
}
```

ãƒãƒƒã‚¯ã‚¨ãƒ³ãƒ‰ã‚‚åŒæ§˜ã« `username` ã«å¤‰æ›´ã—ã¦ãã ã•ã„ã€‚

---

### 3.2 èªè¨¼ API ã‚¨ãƒ³ãƒ‰ãƒã‚¤ãƒ³ãƒˆ

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/entities/auth/api/auth-api.ts`

```typescript
const authApi = {
  async login(credentials: LoginRequest): Promise<LoginResponse> {
    const response = await httpClient.post<LoginResponse>(
      "/auth/login",
      credentials
    );
    return response.data;
  },

  async logout(): Promise<void> {
    await httpClient.post("/auth/logout");
  },
};
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

ãƒãƒƒã‚¯ã‚¨ãƒ³ãƒ‰ã®ã‚¨ãƒ³ãƒ‰ãƒã‚¤ãƒ³ãƒˆæ§‹é€ ã«åˆã‚ã›ã¦å¤‰æ›´ï¼š

```typescript
const authApi = {
  async login(credentials: LoginRequest): Promise<LoginResponse> {
    const response = await httpClient.post<LoginResponse>(
      "/api/v1/auth/login", // â† ãƒãƒ¼ã‚¸ãƒ§ãƒ³ä»˜ãAPIã«å¤‰æ›´
      credentials
    );
    return response.data;
  },

  async logout(): Promise<void> {
    await httpClient.post("/api/v1/auth/logout");
  },
};
```

---

### 3.3 ãƒ¦ãƒ¼ã‚¶ãƒ¼å‹å®šç¾©

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/entities/auth/model/types.ts`

```typescript
export interface User {
  id: number;
  email?: string;
  name?: string;
}
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹: ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã‚’è¿½åŠ 

ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆã®è¦ä»¶ã«å¿œã˜ã¦å¿…è¦ãªãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã‚’è¿½åŠ ï¼š

```typescript
export interface User {
  id: number;
  email: string;
  name: string;
  avatar?: string; // ãƒ—ãƒ­ãƒ•ã‚£ãƒ¼ãƒ«ç”»åƒ
  role: "admin" | "user"; // ãƒ¦ãƒ¼ã‚¶ãƒ¼ãƒ­ãƒ¼ãƒ«
  company?: string; // æ‰€å±ä¼æ¥­
  createdAt: string; // ç™»éŒ²æ—¥æ™‚
}
```

ãƒãƒƒã‚¯ã‚¨ãƒ³ãƒ‰ã®ãƒ¢ãƒ‡ãƒ«ã‚‚åŒæ§˜ã«æ›´æ–°ï¼š

```python
# backend/app/domain/entities/user.py
from pydantic import BaseModel, Field

class User(BaseModel):
    """ãƒ¦ãƒ¼ã‚¶ãƒ¼ã‚¨ãƒ³ãƒ†ã‚£ãƒ†ã‚£"""
    id: int
    email: str
    name: str
    avatar: str | None = None
    role: str = 'user'
    company: str | None= None
    created_at: datetime = field(default_factory=datetime.now)

    class Config:
        """Pydanticè¨­å®š"""

        from_attributes = True  # SQLAlchemyãƒ¢ãƒ‡ãƒ«ã‹ã‚‰å¤‰æ›å¯èƒ½ã«ã™ã‚‹

```

---

### 3.4 èªè¨¼å¾Œã®ãƒªãƒ€ã‚¤ãƒ¬ã‚¯ãƒˆå…ˆ

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/features/auth/login/lib/use-login.ts`

```typescript
onSuccess: (data) => {
  dispatch(setUser({ id: data.user_id }));
  router.push("/dashboard"); // â† ã“ã“
};
```

---

### 3.5 ä¿è­·ãƒ«ãƒ¼ãƒˆã®è¨­å®š

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/middleware.ts`

```typescript
const PROTECTED_PATHS = ["/dashboard"];
const AUTH_PATHS = ["/login"];
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆã®è¦ä»¶ã«å¿œã˜ã¦ãƒ«ãƒ¼ãƒˆã‚’è¿½åŠ ï¼š

```typescript
// èªè¨¼ãŒå¿…è¦ãªãƒšãƒ¼ã‚¸
const PROTECTED_PATHS = [
  "/dashboard",
  "/profile",
  "/settings",
  "/admin",
  "/projects",
];

// ãƒ­ã‚°ã‚¤ãƒ³æ¸ˆã¿ã®å ´åˆã‚¢ã‚¯ã‚»ã‚¹ã§ããªã„ãƒšãƒ¼ã‚¸
const AUTH_PATHS = ["/login", "/register", "/forgot-password"];
```

#### ç‰¹å®šã®ãƒ«ãƒ¼ãƒˆã‚’ç®¡ç†è€…ã®ã¿ã«ã™ã‚‹

```typescript
// middleware.ts
export function middleware(request: NextRequest) {
  const pathname = request.nextUrl.pathname;
  const accessToken = request.cookies.get("access_token");

  // ç®¡ç†è€…å°‚ç”¨ãƒ«ãƒ¼ãƒˆ
  if (pathname.startsWith("/admin")) {
    if (!accessToken) {
      return NextResponse.redirect(new URL("/login", request.url));
    }

    // TODO: ãƒˆãƒ¼ã‚¯ãƒ³ã‹ã‚‰ãƒ­ãƒ¼ãƒ«ã‚’å–å¾—ã—ã¦ç¢ºèª
    // const user = decodeToken(accessToken);
    // if (user.role !== 'admin') {
    //   return NextResponse.redirect(new URL('/dashboard', request.url));
    // }
  }

  // é€šå¸¸ã®ä¿è­·ãƒ«ãƒ¼ãƒˆ
  if (PROTECTED_PATHS.some((path) => pathname.startsWith(path))) {
    if (!accessToken) {
      return NextResponse.redirect(new URL("/login", request.url));
    }
  }

  return NextResponse.next();
}
```

---

### 3.6 Cookie/Token è¨­å®š

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `backend/app/presentation/api/auth_api.py`

```python
response.set_cookie(
    key='access_token',
    value=output_dto.access_token,
    httponly=True,
    secure=True,
    samesite='lax',
    max_age=7 * 24 * 60 * 60,  # 7æ—¥é–“
)
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹ 1: æœ‰åŠ¹æœŸé™ã‚’ 1 æ—¥ã«å¤‰æ›´

```python
response.set_cookie(
    key='access_token',
    value=output_dto.access_token,
    httponly=True,
    secure=True,
    samesite='lax',
    max_age=24 * 60 * 60,  # 1æ—¥
)
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹ 2: Cookie åã‚’ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆå›ºæœ‰ã«

```python
response.set_cookie(
    key='yourapp_token',  # â† ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆå›ºæœ‰ã®åå‰
    value=output_dto.access_token,
    httponly=True,
    secure=True,
    samesite='lax',
    max_age=7 * 24 * 60 * 60,
)
```

**æ³¨æ„**: Frontend ã® middleware ã§ã‚‚ Cookie åã‚’å¤‰æ›´ã™ã‚‹å¿…è¦ãŒã‚ã‚Šã¾ã™ã€‚

```typescript
// frontend/src/middleware.ts
const accessToken = request.cookies.get("yourapp_token"); // â† å¤‰æ›´
```

---

## 4. è‰²ãƒ»ãƒ†ãƒ¼ãƒ

### 4.1 ã‚°ãƒ­ãƒ¼ãƒãƒ« CSS ã®è‰²å®šç¾©

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/app/globals.css`

```css
@layer base {
  :root {
    --primary: 222.2 47.4% 11.2%;
    --primary-foreground: 210 40% 98%;
    --secondary: 210 40% 96.1%;
    --accent: 210 40% 96.1%;
    --destructive: 0 84.2% 60.2%;
    --border: 214.3 31.8% 91.4%;
    --radius: 0.5rem;
  }
}
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

**Step 1**: ãƒ–ãƒ©ãƒ³ãƒ‰ã‚«ãƒ©ãƒ¼ã‚’æ±ºå®š

[HSL Color Picker](https://hslpicker.com/) ã‚’ä½¿ã£ã¦è‰²ã‚’é¸æŠã€‚

**ä¾‹**: ãƒ–ãƒ«ãƒ¼ç³»ã®ãƒ–ãƒ©ãƒ³ãƒ‰ã‚«ãƒ©ãƒ¼

```css
:root {
  /* ãƒ—ãƒ©ã‚¤ãƒãƒªã‚«ãƒ©ãƒ¼ï¼ˆãƒ–ãƒ©ãƒ³ãƒ‰ã‚«ãƒ©ãƒ¼ï¼‰ */
  --primary: 210 100% 50%; /* é®®ã‚„ã‹ãªãƒ–ãƒ«ãƒ¼ */
  --primary-foreground: 0 0% 100%; /* ç™½æ–‡å­— */

  /* ã‚»ã‚«ãƒ³ãƒ€ãƒªã‚«ãƒ©ãƒ¼ */
  --secondary: 210 40% 96%; /* è–„ã„ãƒ–ãƒ«ãƒ¼ */
  --secondary-foreground: 222.2 47.4% 11.2%;

  /* ã‚¢ã‚¯ã‚»ãƒ³ãƒˆã‚«ãƒ©ãƒ¼ */
  --accent: 280 100% 50%; /* ç´« */
  --accent-foreground: 0 0% 100%;

  /* ãã®ä»– */
  --destructive: 0 84.2% 60.2%; /* èµ¤ï¼ˆå‰Šé™¤ãƒœã‚¿ãƒ³ç­‰ï¼‰ */
  --border: 214.3 31.8% 91.4%;
  --radius: 0.5rem;
}
```

**Step 2**: ãƒ€ãƒ¼ã‚¯ãƒ¢ãƒ¼ãƒ‰ã®èª¿æ•´

```css
.dark {
  --primary: 210 100% 60%;
  --primary-foreground: 222.2 47.4% 11.2%;
  --secondary: 217.2 32.6% 17.5%;
  --secondary-foreground: 210 40% 98%;
  --accent: 280 100% 60%;
  --accent-foreground: 0 0% 100%;
}
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹: ã‚«ã‚¹ã‚¿ãƒ ã‚«ãƒ©ãƒ¼ã®è¿½åŠ 

```css
:root {
  /* æ—¢å­˜ã®ã‚«ãƒ©ãƒ¼ */
  --primary: 210 100% 50%;

  /* ã‚«ã‚¹ã‚¿ãƒ ã‚«ãƒ©ãƒ¼ */
  --brand-blue: 210 100% 50%;
  --brand-green: 142 76% 36%;
  --success: 142 76% 36%;
  --warning: 38 92% 50%;
  --info: 199 89% 48%;
}
```

**Tailwind ã§ä½¿ç”¨**:

```typescript
// tailwind.config.ts
export default {
  theme: {
    extend: {
      colors: {
        brand: {
          blue: "hsl(var(--brand-blue))",
          green: "hsl(var(--brand-green))",
        },
        success: "hsl(var(--success))",
        warning: "hsl(var(--warning))",
        info: "hsl(var(--info))",
      },
    },
  },
};
```

```tsx
// ä½¿ç”¨ä¾‹
<Button className="bg-success">ä¿å­˜</Button>
<Alert className="bg-warning">è­¦å‘Šãƒ¡ãƒƒã‚»ãƒ¼ã‚¸</Alert>
```

---

### 4.2 Tailwind è¨­å®š

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/tailwind.config.ts`

```typescript
export default {
  theme: {
    extend: {
      colors: {
        primary: {
          DEFAULT: "hsl(var(--primary))",
          foreground: "hsl(var(--primary-foreground))",
        },
        // ...
      },
    },
  },
};
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹ 1: ã‚«ã‚¹ã‚¿ãƒ ãƒ•ã‚©ãƒ³ãƒˆã®è¿½åŠ 

```typescript
// tailwind.config.ts
export default {
  theme: {
    extend: {
      fontFamily: {
        sans: ["Inter", "Noto Sans JP", "sans-serif"],
        heading: ["Montserrat", "sans-serif"],
      },
    },
  },
};
```

```tsx
// ä½¿ç”¨ä¾‹
<h1 className="font-heading">è¦‹å‡ºã—</h1>
<p className="font-sans">æœ¬æ–‡</p>
```

---

### 4.3 shadcn/ui ã®ãƒ†ãƒ¼ãƒ

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/components.json`

```json
{
  "style": "default",
  "tailwind": {
    "baseColor": "slate",
    "cssVariables": true
  }
}
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

`baseColor` ã‚’å¤‰æ›´ã—ã¦ãƒ™ãƒ¼ã‚¹ã‚«ãƒ©ãƒ¼ã‚’èª¿æ•´ï¼š

```json
{
  "style": "default",
  "tailwind": {
    "baseColor": "zinc", // slate, gray, zinc, neutral, stone
    "cssVariables": true
  }
}
```

**å¤‰æ›´å¾Œã¯ shadcn/ui ã‚³ãƒ³ãƒãƒ¼ãƒãƒ³ãƒˆã‚’å†è¿½åŠ **:

```bash
npx shadcn@latest add button
npx shadcn@latest add card
```

---

## 5. ãƒšãƒ¼ã‚¸ãƒ»ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ

### 5.1 ãƒ­ã‚°ã‚¤ãƒ³ãƒšãƒ¼ã‚¸

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/page-components/login/ui/LoginPage.tsx`

```tsx
export function LoginPage() {
  return (
    <div className="flex min-h-screen items-center justify-center">
      <LoginForm />
    </div>
  );
}
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹ 1: ãƒ­ã‚´ã¨èƒŒæ™¯ã‚’è¿½åŠ 

```tsx
export function LoginPage() {
  return (
    <div className="flex min-h-screen items-center justify-center bg-gradient-to-br from-blue-500 to-purple-600">
      <div className="w-full max-w-md space-y-8">
        {/* ãƒ­ã‚´ */}
        <div className="text-center">
          <img src="/logo.png" alt="Logo" className="mx-auto h-12 w-auto" />
          <h2 className="mt-6 text-3xl font-bold text-white">ã‚ˆã†ã“ã</h2>
        </div>

        {/* ãƒ­ã‚°ã‚¤ãƒ³ãƒ•ã‚©ãƒ¼ãƒ  */}
        <LoginForm />

        {/* ãƒ•ãƒƒã‚¿ãƒ¼ãƒªãƒ³ã‚¯ */}
        <div className="text-center text-sm text-white">
          <a href="/register" className="hover:underline">
            æ–°è¦ç™»éŒ²ã¯ã“ã¡ã‚‰
          </a>
        </div>
      </div>
    </div>
  );
}
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹ 2: 2 ã‚«ãƒ©ãƒ ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ

```tsx
export function LoginPage() {
  return (
    <div className="flex min-h-screen">
      {/* å·¦å´: ãƒ–ãƒ©ãƒ³ãƒ‡ã‚£ãƒ³ã‚°ã‚¨ãƒªã‚¢ */}
      <div className="hidden w-1/2 bg-primary p-12 lg:block">
        <div className="flex h-full flex-col justify-between">
          <img src="/logo-white.png" alt="Logo" className="h-8 w-auto" />
          <div>
            <h1 className="text-4xl font-bold text-white">
              ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆã‚’åŠ é€Ÿã™ã‚‹
            </h1>
            <p className="mt-4 text-xl text-white/80">
              æœ€é«˜ã®ã‚½ãƒªãƒ¥ãƒ¼ã‚·ãƒ§ãƒ³ã‚’ã‚ãªãŸã«
            </p>
          </div>
          <div className="text-sm text-white/60">Â© 2025 Your Company</div>
        </div>
      </div>

      {/* å³å´: ãƒ­ã‚°ã‚¤ãƒ³ãƒ•ã‚©ãƒ¼ãƒ  */}
      <div className="flex w-full items-center justify-center lg:w-1/2">
        <div className="w-full max-w-md px-8">
          <LoginForm />
        </div>
      </div>
    </div>
  );
}
```

---

### 5.2 ãƒ€ãƒƒã‚·ãƒ¥ãƒœãƒ¼ãƒ‰ãƒšãƒ¼ã‚¸

ã“ã“ã¯å¥½ãã«å¤‰ãˆã¦ãã ã•ã„

---

### 5.3 èªè¨¼ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆ

ãƒ˜ãƒƒãƒ€ãƒ¼ã‚„ã‚µã‚¤ãƒ‰ãƒãƒ¼ãªã©ã€å¿…è¦ã«å¿œã˜ã¦ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆã‚’è¿½åŠ ã—ã¦ãã ã•ã„ã€‚

---

## 6. ç’°å¢ƒè¨­å®š

### 6.1 ã‚¢ãƒ—ãƒªåãƒ»ãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/app/layout.tsx`

```typescript
export const metadata: Metadata = {
  title: "AI Solution Template",
  description: "Startup template with Next.js and FastAPI",
};
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

```typescript
export const metadata: Metadata = {
  title: "Your App Name",
  description: "Your app description for SEO",
  keywords: ["keyword1", "keyword2", "keyword3"],
  authors: [{ name: "Your Company" }],
  openGraph: {
    title: "Your App Name",
    description: "Your app description",
    url: "https://yourapp.com",
    siteName: "Your App Name",
    images: [
      {
        url: "https://yourapp.com/og-image.png",
        width: 1200,
        height: 630,
      },
    ],
    locale: "ja_JP",
    type: "website",
  },
  twitter: {
    card: "summary_large_image",
    title: "Your App Name",
    description: "Your app description",
    images: ["https://yourapp.com/twitter-image.png"],
  },
};
```

**å…¬é–‹ãƒšãƒ¼ã‚¸ã®ãƒ¬ã‚¤ã‚¢ã‚¦ãƒˆã‚‚æ›´æ–°**:

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/src/app/(public)/layout.tsx`

```typescript
export const metadata: Metadata = {
  title: "Your App - Login",
  description: "Login to Your App",
};
```

---

### 6.2 ç’°å¢ƒå¤‰æ•°

#### Frontend ç’°å¢ƒå¤‰æ•°

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/.env.local.example`

```bash
# ç¾åœ¨ã®è¨­å®š
NEXT_PUBLIC_API_URL=http://localhost:8000
NODE_ENV=development
```

**ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹**:

```bash
# APIè¨­å®š
NEXT_PUBLIC_API_URL=https://api.yourapp.com

# ã‚¢ãƒ—ãƒªã‚±ãƒ¼ã‚·ãƒ§ãƒ³è¨­å®š
NEXT_PUBLIC_APP_NAME=YourAppName
NEXT_PUBLIC_APP_VERSION=1.0.0

# å¤–éƒ¨ã‚µãƒ¼ãƒ“ã‚¹
NEXT_PUBLIC_GOOGLE_ANALYTICS_ID=G-XXXXXXXXXX
NEXT_PUBLIC_SENTRY_DSN=your-sentry-dsn

# æ©Ÿèƒ½ãƒ•ãƒ©ã‚°
NEXT_PUBLIC_ENABLE_FEATURE_X=true

# ç’°å¢ƒ
NODE_ENV=production
```

**`.env.local` ãƒ•ã‚¡ã‚¤ãƒ«ã‚’ä½œæˆ**:

```bash
cp .env.local.example .env.local
# .env.local ã‚’ç·¨é›†
```

#### Backend ç’°å¢ƒå¤‰æ•°

**ãƒ•ã‚¡ã‚¤ãƒ«**: `backend/.env.example`

```bash
# ç¾åœ¨ã®è¨­å®š
# Application
ENVIRONMENT=development

# Database
POSTGRES_USER=app_user
POSTGRES_PASSWORD=app_password
POSTGRES_DB=ai_solution_db
POSTGRES_HOST=db
POSTGRES_PORT=5432

# Database URL for SQLAlchemy
DATABASE_URL=postgresql+psycopg2://app_user:app_password@db:5432/ai_solution_db

# JWT Settings (RS256)
# RSAéµãƒšã‚¢ã‚’ç”Ÿæˆã™ã‚‹ã«ã¯: make generate-rsa-keys
# æ”¹è¡Œã¯ \n ã«å¤‰æ›ã—ã¦ãã ã•ã„
JWT_ALGORITHM=RS256
JWT_EXPIRATION_HOURS=24
JWT_PRIVATE_KEY=your-private-key-here
JWT_PUBLIC_KEY=your-public-key-here
```

**ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºä¾‹**:

```bash
# ãƒ‡ãƒ¼ã‚¿ãƒ™ãƒ¼ã‚¹è¨­å®š
POSTGRES_USER=yourapp_user
POSTGRES_PASSWORD=your_secure_password_123
POSTGRES_DB=yourapp_production
DATABASE_URL=postgresql://yourapp_user:your_secure_password_123@db:5432/yourapp_production

# JWTè¨­å®šï¼ˆå¾Œè¿°ï¼‰
JWT_PRIVATE_KEY=<ç”Ÿæˆã—ãŸç§˜å¯†éµ>
JWT_PUBLIC_KEY=<ç”Ÿæˆã—ãŸå…¬é–‹éµ>
JWT_ALGORITHM=RS256

# ã‚¢ãƒ—ãƒªã‚±ãƒ¼ã‚·ãƒ§ãƒ³è¨­å®š
APP_NAME=YourApp
APP_VERSION=1.0.0
DEBUG=false

# CORSè¨­å®š
ALLOWED_ORIGINS=https://yourapp.com,https://www.yourapp.com

# å¤–éƒ¨ã‚µãƒ¼ãƒ“ã‚¹
SENDGRID_API_KEY=your_sendgrid_key
AWS_ACCESS_KEY_ID=your_aws_key
AWS_SECRET_ACCESS_KEY=your_aws_secret
```

**`.env` ãƒ•ã‚¡ã‚¤ãƒ«ã‚’ä½œæˆ**:

```bash
cd backend
cp .env.example .env
# .env ã‚’ç·¨é›†
```

---

### 6.3 JWT éµã®è¨­å®š

#### JWT éµã®ç”Ÿæˆ

- ä¸‹è¨˜ã‚³ãƒãƒ³ãƒ‰ã§ã€ç§˜å¯†éµãƒ»å…¬é–‹éµã®ãƒšã‚¢ãŒæ–°ã—ãç”Ÿæˆã•ã‚Œã‚‹ã€‚
- ã“ã‚Œã‚‰ã®å†…å®¹ã‚’ã‚³ãƒ”ãƒ¼ã—ã€.env ã«æ›¸ãè¾¼ã‚€å¿…è¦ãŒã‚ã‚‹ã€‚

```bash
make generate-rsa-keys
```

**æ³¨æ„**:

- ç§˜å¯†éµã¯çµ¶å¯¾ã« Git ã«ã‚³ãƒŸãƒƒãƒˆã—ãªã„
- æœ¬ç•ªç’°å¢ƒã§ã¯ã€æ–°ã—ãéµã®ãƒšã‚¢ã‚’ç”Ÿæˆã—ã€ç’°å¢ƒå¤‰æ•°ã¾ãŸã¯ã‚·ãƒ¼ã‚¯ãƒ¬ãƒƒãƒˆç®¡ç†ã‚µãƒ¼ãƒ“ã‚¹ï¼ˆAWS Secrets Manager ç­‰ï¼‰ã‚’ä½¿ç”¨ã—ã¦ç®¡ç†ã™ã‚‹ã¹ãã€‚

---

### 6.4 CORS è¨­å®š

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `backend/app/main.py`

```python
allowed_origins = [
    'http://localhost:3000',
    'http://127.0.0.1:3000',
    # ã“ã“ã«æœ¬ç•ªç’°å¢ƒã®ãƒ‰ãƒ¡ã‚¤ãƒ³ã‚’ãƒ‡ãƒ—ãƒ­ã‚¤å¾Œè¿½åŠ ã™ã‚‹ã€‚
]
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

æœ¬ç•ªç’°å¢ƒã®ãƒ‰ãƒ¡ã‚¤ãƒ³ã‚’è¿½åŠ ï¼š

```python
allowed_origins = [
    'http://localhost:3000',           # ãƒ­ãƒ¼ã‚«ãƒ«é–‹ç™º
    'http://127.0.0.1:3000',
    'https://yourapp.com',             # æœ¬ç•ªç’°å¢ƒ
    'https://www.yourapp.com',
    'https://staging.yourapp.com',     # ã‚¹ãƒ†ãƒ¼ã‚¸ãƒ³ã‚°ç’°å¢ƒ
]
```

---

### 6.5 ãƒ­ã‚´ãƒ»ãƒ•ã‚¡ãƒ“ã‚³ãƒ³

#### ç¾åœ¨ã®ãƒ•ã‚¡ã‚¤ãƒ«

**ãƒ‡ã‚£ãƒ¬ã‚¯ãƒˆãƒª**: `frontend/public/`

- `favicon.ico`
- `apple-icon.png`
- `logo.png`

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

**Step 1**: ãƒ­ã‚´ãƒ»ã‚¢ã‚¤ã‚³ãƒ³ã‚’æº–å‚™

ä»¥ä¸‹ã®ã‚µã‚¤ã‚ºã‚’ç”¨æ„ï¼š

- **favicon.ico**: 16x16, 32x32
- **apple-icon.png**: 180x180ï¼ˆiOS ãƒ›ãƒ¼ãƒ ç”»é¢ç”¨ï¼‰
- **logo.png**: ãƒ˜ãƒƒãƒ€ãƒ¼ç­‰ã§ä½¿ç”¨ï¼ˆæ¨å¥¨: 200x50 ç¨‹åº¦ï¼‰
- **og-image.png**: OGP ç”¨ï¼ˆæ¨å¥¨: 1200x630ï¼‰

**Step 2**: ãƒ•ã‚¡ã‚¤ãƒ«ã‚’ç½®ãæ›ãˆ

```bash
# frontend/public/ ã«ãƒ•ã‚¡ã‚¤ãƒ«ã‚’ã‚³ãƒ”ãƒ¼
cp your-favicon.ico public/favicon.ico
cp your-apple-icon.png public/apple-icon.png
cp your-logo.png public/logo.png
cp your-og-image.png public/og-image.png
```

**Step 3**: ãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿ã‚’æ›´æ–°

```typescript
// frontend/src/app/layout.tsx
export const metadata: Metadata = {
  title: "Your App",
  description: "Your app description",
  icons: {
    icon: "/favicon.ico",
    apple: "/apple-icon.png",
  },
  openGraph: {
    images: ["/og-image.png"],
  },
};
```

**Step 4**: ãƒ­ã‚´ã‚’ä½¿ç”¨

```tsx
// ã‚³ãƒ³ãƒãƒ¼ãƒãƒ³ãƒˆå†…ã§ä½¿ç”¨
<img src="/logo.png" alt="Your App Logo" className="h-8" />
```

---

### 6.6 package.json

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `frontend/package.json`

```json
{
  "name": "ai-solution-template-frontend",
  "version": "0.1.0",
  "private": true
}
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

```json
{
  "name": "yourapp-frontend",
  "version": "1.0.0",
  "description": "Your app description",
  "author": "Your Company",
  "license": "MIT",
  "private": true
}
```

---

## 7. ã‚¤ãƒ³ãƒ•ãƒ©è¨­å®š

### 7.1 CDK ã‚·ã‚¹ãƒ†ãƒ å

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `infra/bin/solution-project-template.ts`

```typescript
const systemName = "sol";
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆå›ºæœ‰ã®ã‚·ã‚¹ãƒ†ãƒ åã«å¤‰æ›´ï¼ˆ3-5 æ–‡å­—ã®å°æ–‡å­—æ¨å¥¨ï¼‰:

```typescript
const systemName = "yourapp"; // ã¾ãŸã¯ 'yap', 'myapp' ãªã©
```

**æ³¨æ„**: ã“ã®åå‰ã¯ AWS ãƒªã‚½ãƒ¼ã‚¹åã®ãƒ—ãƒ¬ãƒ•ã‚£ãƒƒã‚¯ã‚¹ã«ãªã‚Šã¾ã™ã€‚

ä¾‹: `yourapp-dev-vpc`, `yourapp-prod-ecs-cluster`

---

### 7.2 Docker Compose

#### ç¾åœ¨ã®å®Ÿè£…

**ãƒ•ã‚¡ã‚¤ãƒ«**: `docker-compose.yml`

```yaml
services:
  db:
    container_name: postgres_container_ai_solution
    # ...

  pgadmin:
    container_name: pgadmin_container_ai_solution
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@example.com
      PGADMIN_DEFAULT_PASSWORD: admin
```

#### ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºæ–¹æ³•

ã‚³ãƒ³ãƒ†ãƒŠåã¨ pgAdmin èªè¨¼æƒ…å ±ã‚’å¤‰æ›´ï¼š

```yaml
services:
  db:
    container_name: postgres_yourapp
    # ...

  pgadmin:
    container_name: pgadmin_yourapp
    environment:
      PGADMIN_DEFAULT_EMAIL: your-email@example.com
      PGADMIN_DEFAULT_PASSWORD: your_secure_password
```

---

## 8. ãƒã‚§ãƒƒã‚¯ãƒªã‚¹ãƒˆ

ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºå®Œäº†ãƒã‚§ãƒƒã‚¯ãƒªã‚¹ãƒˆ

- [ ] ãƒ­ã‚°ã‚¤ãƒ³ãƒ•ã‚£ãƒ¼ãƒ«ãƒ‰ã‚’å¤‰æ›´ï¼ˆlogin_id â†’ email ç­‰ï¼‰
- [ ] èªè¨¼ API ã‚¨ãƒ³ãƒ‰ãƒã‚¤ãƒ³ãƒˆã‚’ç¢ºèªãƒ»å¤‰æ›´
- [ ] ãƒ¦ãƒ¼ã‚¶ãƒ¼å‹å®šç¾©ã‚’ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆã«åˆã‚ã›ã¦å¤‰æ›´
- [ ] ã‚¢ãƒ—ãƒªåãƒ»ãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿ã‚’å¤‰æ›´
- [ ] Frontend ç’°å¢ƒå¤‰æ•°ã‚’è¨­å®šï¼ˆ`.env`ï¼‰
- [ ] Backend ç’°å¢ƒå¤‰æ•°ã‚’è¨­å®šï¼ˆ`.env`ï¼‰
- [ ] CORS è¨­å®šã«æœ¬ç•ªãƒ‰ãƒ¡ã‚¤ãƒ³ã‚’è¿½åŠ 
- [ ] JWT éµã‚’ç”Ÿæˆãƒ»è¨­å®š
- [ ] ãƒ­ã‚´ãƒ»ãƒ•ã‚¡ãƒ“ã‚³ãƒ³ã‚’å·®ã—æ›¿ãˆ
- [ ] ãƒ—ãƒ©ã‚¤ãƒãƒªã‚«ãƒ©ãƒ¼ã‚’å¤‰æ›´ï¼ˆ`globals.css`ï¼‰
- [ ] ãƒ€ãƒƒã‚·ãƒ¥ãƒœãƒ¼ãƒ‰ãƒšãƒ¼ã‚¸ã‚’ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚º
- [ ] ãƒ­ã‚°ã‚¤ãƒ³ãƒšãƒ¼ã‚¸ã®ãƒ‡ã‚¶ã‚¤ãƒ³ã‚’å¤‰æ›´
- [ ] ä¿è­·ãƒ«ãƒ¼ãƒˆã‚’è¿½åŠ 
- [ ] èªè¨¼å¾Œã®ãƒªãƒ€ã‚¤ãƒ¬ã‚¯ãƒˆå…ˆã‚’ç¢ºèª
- [ ] CDK ã‚·ã‚¹ãƒ†ãƒ åã‚’å¤‰æ›´
- [ ] Docker ã‚³ãƒ³ãƒ†ãƒŠåã‚’å¤‰æ›´
- [ ] pgAdmin èªè¨¼æƒ…å ±ã‚’å¤‰æ›´
- [ ] ã‚«ã‚¹ã‚¿ãƒ ã‚«ãƒ©ãƒ¼ã‚’è¿½åŠ 
- [ ] Tailwind ã‚«ã‚¹ã‚¿ãƒ è¨­å®šã‚’è¿½åŠ 

---

## ã¾ã¨ã‚

ã“ã®ã‚¬ã‚¤ãƒ‰ã«å¾“ã£ã¦ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã‚’é€²ã‚ã‚‹ã“ã¨ã§ã€ãƒ†ãƒ³ãƒ—ãƒ¬ãƒ¼ãƒˆã‚’ãƒ—ãƒ­ã‚¸ã‚§ã‚¯ãƒˆå›ºæœ‰ã®è¦ä»¶ã«åˆã‚ã›ã‚‹ã“ã¨ãŒã§ãã¾ã™ã€‚

### ã‚µãƒãƒ¼ãƒˆ

ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã§å›°ã£ãŸå ´åˆã¯ã€ä»¥ä¸‹ã®ãƒ‰ã‚­ãƒ¥ãƒ¡ãƒ³ãƒˆã‚’å‚ç…§ã—ã¦ãã ã•ã„ï¼š

- [FRONTEND_ARCHITECTURE.md](./FRONTEND_ARCHITECTURE.md) - ã‚¢ãƒ¼ã‚­ãƒ†ã‚¯ãƒãƒ£è¨­è¨ˆ
- [FRONTEND_DEVELOPMENT_GUIDE.md](./FRONTEND_DEVELOPMENT_GUIDE.md) - é–‹ç™ºã‚¬ã‚¤ãƒ‰
- [FRONTEND_TESTING_GUIDE.md](./FRONTEND_TESTING_GUIDE.md) - ãƒ†ã‚¹ãƒˆã‚¬ã‚¤ãƒ‰

---

**æœ€çµ‚æ›´æ–°**: 2025-11-12
**ãƒãƒ¼ã‚¸ãƒ§ãƒ³**: 1.0.0
